File: c/leap.cpp
//program to determine leap year or not


#include <stdio.h>
#include<conio.h>
 
int main()
{
  int year;
 
  printf("Enter a year to check if it is a leap year\n");
  scanf("%d", &year);
  if(year%4 == 0) 
  {
            
  if( year%100 == 0)
  { 
  if ( year%400 == 0)
  printf("%d is a leap year.\n", year);
  else 
  printf("it is not a leap year");
  }
  else
  printf("it is a leap year");
  }
  else
  {
  printf("it is not a leap year");
  }
  getch();
}




    

File: java/leap.java

import java.util.Scanner;
public class leap{
public static void main(String args[])
{
int year;
System.out.println("enter the year");
Scanner in = new Scanner(System.in);
 
year= in.nextInt();

le(year);
}
public static  void le(int a)
{
if(a%4 == 0) 
  {
            
  if( a%100 == 0)
  { 
  if ( a%400 == 0)
  System.out.println("it is a leap year.\n");
  else 
  System.out.println("it is not a leap year");
  }
  else
 System.out.println("it is a leap year");
  }
  else
  {
  System.out.println("it is not a leap year");
}
}
}